src/apps/reservation-list/modal/pause-reservation/pause-reservation.tsx

```diff
@@ -0,0 +1,123 @@
+import React, { FC, useCallback, useState, useEffect } from "react";
+import { Link } from "../../../../components/atoms/link";
+import Modal, { useModalButtonHandler } from "../../../../core/utils/modal";
+import { useText } from "../../../../core/utils/text";
+import DateInput from "./date-input";
+import { useUpdateV5 } from "../../../../core/fbs/fbs";
+import { PatronV5 } from "../../../../core/fbs/model";
+import { getModalIds } from "../../../../core/utils/helpers/general";
+import { useConfig } from "../../../../core/utils/config";
+
+interface PauseReservationProps {
+  id: string;
+  user: PatronV5;
+}
+
+const PauseReservation: FC<PauseReservationProps> = ({ id, user }) => {
+  const t = useText();
+  const { mutate } = useUpdateV5();
+  const { close } = useModalButtonHandler();
+  const { pauseReservation } = getModalIds();
+  const config = useConfig();
+  const [startDate, setStartDate] = useState<string>(
+    (config("pauseReservationStartDateConfig") as string) || ""
+  );
+  const [endDate, setEndDate] = useState<string | null>("");
+
+  const save = useCallback(() => {
+    if (user && startDate && endDate) {
+      mutate(
+        {
+          data: {
+            patron: {
+              preferredPickupBranch: user.preferredPickupBranch,
+              receiveEmail: user.receiveEmail,
+              receivePostalMail: user.receivePostalMail,
+              receiveSms: user.receiveSms,
+              onHold: { from: startDate, to: endDate }
+            }
+          }
+        },
+        {
+          onSuccess: () => {
+            close(pauseReservation as string);
+          },
+          // todo error handling, missing in figma
+          onError: () => {}
+        }
+      );
+    }
+  }, [close, endDate, mutate, pauseReservation, startDate, user]);
+
+  useEffect(() => {
+    if (user?.onHold?.from) {
+      setStartDate(user.onHold.from);
+    }
+    if (user?.onHold?.to) {
+      setEndDate(user.onHold.to);
+    }
+  }, [user?.onHold]);
```

Comment:
seems weird to me with optional chaining in the dependency array. But I guess it works? :)